ang/crmMailing/BlockPreview.js

```diff
@@ -29,33 +29,35 @@
         scope.previewTestGroup = function(e) {
           var $dialog = $(this);
           $dialog.html('<div class="crm-loading-element"></div>').parent().find('button[data-op=yes]').prop('disabled', true);
-          $dialog.dialog('option', 'title', ts('Send to %1', {1: _.pluck(_.where(scope.crmMailingConst.testGroupNames, {id: scope.testGroup.gid}), 'title')[0]}));
-          CRM.api3('contact', 'get', {
-            group: scope.testGroup.gid,
-            options: {limit: 0},
-            return: 'display_name,email'
-          }).done(function(data) {
-            var count = 0,
-            // Fixme: should this be in a template?
+          CRM.api3('group', 'getsingle', {id: scope.testGroup.gid, return: 'title'}).done(function(group) {
+            $dialog.dialog('option', 'title', ts('Send to %1', {1: group.title}));
```

Comment:
By adding to the github url `?w=1` to filter out whitespace changes I was able to understand this change :)
It looks like you're adding an api call to lookup the group name, but that data hasn't actually been removed from the scope so I don't think changes to this file are necessary.

Reply:
@colemanw yes if you notice in old line 32 we are plucking out the title from an array ids, titles of groups. In this case we are no longer using that array (array is `scope.crmMailingConst.testGroupNames` so we have to because all we have then at present is the id of the group we have to make the API call to get the title of the group.  to set as part of the title of the dialog

Reply:
My point was that `scope.crmMailingConst.testGroupNames` hasn't actually been removed by this PR, so it should still be accessible to this function, no?

Reply:
The long term goal i see is getting rid of `scope.crmMailingConst.testGroupNames` eventually to reduce the bloat in the DOM, We started by initially getting agileware to convert the recipients box to be AJAX, now we're fixing the test group stuff here and then need to solve the unsubscribe group. Once all that is done and check that mosaico gets appropriate changes. We can then remove `testGroupNames` out so that we reduce the amount of data getting pushed into the DOM in the boot. On large sites it has been found the loading of this array and also the message templates slows it down significantly (message template stuff was taken care of in a separate PR)

Reply:
Ok. Technically that information *is* in memory and doesn't have to be fetched from the api, because the Select2 element has already retrieved the group title. Let me think about how best to access it to avoid the extra api call.